vowpalwabbit/core/src/reductions/scorer.cc

```diff
@@ -21,6 +22,11 @@ using namespace VW::config;
 
 namespace
 {
+struct options_scorer
+{
+  std::string link;
```

Comment:
it would be good to move away from string repr to another more restrictive type.

Reply:
I considered an enum class, but we still need access to the string itself regardless (for the name in this case..). I'm not sure what the final design of these objects will look like, but so far they all look like structs with public members so I'm sticking with that for now